### PR TITLE
sdn: disable hostports when CRI-O is used

### PR DESCRIPTION
**Pull request by dcbw**
For now...

@openshift/networking @knobunc @rajatchopra

### CODE REVIEW

**Comment by imcsk8** [comment]
LGTM

**Comment by danwinship** [comment]
> The real fix for this is to add hostport handling to Kube's CNI
> driver, but in the "split" mode discussed upstream where kube's
> CNI driver handles the port reservations on the host, while the
> plugin handles the actual iptables rules.  CRIO should be converted
> to this scheme as well

Are there kube and CNI bugs filed about this?
/lgtm

File: pkg/network/node/pod.go [hunk 1]
```diff
@@ -499,8 +504,10 @@ func (m *podManager) setup(req *cniserver.PodRequest) (cnitypes.Result, *running
 	defer func() {
 		if !success {
 			m.ipamDel(req.SandboxID)
-			if err := m.hostportSyncer.SyncHostports(Tun0, m.getRunningPods()); err != nil {
-				glog.Warningf("failed syncing hostports: %v", err)
+			if m.hostportSyncer != nil {
```
Comment (rajatchopra) [inline pkg/network/node/pod.go]: chain of booleans. Can we check for enableHostports here?

File: pkg/network/node/pod.go [hunk 2]
```diff
@@ -511,8 +518,10 @@ func (m *podManager) setup(req *cniserver.PodRequest) (cnitypes.Result, *running
 		return nil, nil, err
 	}
 	podPortMapping := kubehostport.ConstructPodPortMapping(&v1Pod, podIP)
-	if err := m.hostportSyncer.OpenPodHostportsAndSync(podPortMapping, Tun0, m.getRunningPods()); err != nil {
-		return nil, nil, err
+	if m.hostportSyncer != nil {
```
Comment (rajatchopra) [inline pkg/network/node/pod.go]: same as above.

File: pkg/network/node/pod.go [hunk 3]
```diff
@@ -631,8 +640,10 @@ func (m *podManager) teardown(req *cniserver.PodRequest) error {
 		errList = append(errList, err)
 	}
 
-	if err := m.hostportSyncer.SyncHostports(Tun0, m.getRunningPods()); err != nil {
-		errList = append(errList, err)
+	if m.hostportSyncer != nil {
```
Comment (rajatchopra) [inline pkg/network/node/pod.go]: as pointed above

**Comment by dcbw** [comment]
> Are there kube and CNI bugs filed about this?

It's mostly about CRIO.  In yesterday's SIG Net meeting Tim indicated he'd like to punt all this out to runtimes and plugins.  Kube's CNI driver doesn't do anything related to hostports at all.  And CRIO probably shouldn't either, that's what the CNI 'portmap' plugin is for.

CRIO issue filed as https://github.com/kubernetes-incubator/cri-o/issues/941

**Comment by kargakis** [comment]
cc @runcom @mrunalp 

/approve

**Pull request merged**